docs/writing-docs/skillmaps.md

```diff
@@ -0,0 +1,205 @@
+### ~
+
+## Skillmap paths
+
+A skillmap path has a starting point and a sequence of nodes (activities) to progress through. The activities in nodes along the path are locked until the previous node's an activity is finished.
+
+![Individual skillmap path](/static/skillmaps/skillmap-path-intro.jpg)
+
+The first node is unlocked and ready to start.
+
+![Skillmap start node](/static/skillmaps/skillmap-start-node.jpg)
+
+Each node contains a tutorial exercise to teach a certain skill or concept. Completing the tutorial (activity) will unlock the next node along the pathway. 
+
+![Skillmap completed node](/static/skillmaps/skillmap-complete-node.jpg)
+
+After the last tutorial node, the student reaches the end of the path on the map and arrives at the Reward. The Reward node greets the student with an affirming message and awards them a downloadable "Certificate of Completion".
+
+![Skillmap start node](/static/skillmaps/skillmap-reward-node.jpg)
+
+### Skillmap structure
+
+A skillmap is described by a markdown document with sections for the map settings, its paths, and the activities (nodes) in the paths.
+
+```
+# map-id
+
+* mapProperty1: ...
+* mapProperty2: ...
+
+## path-id1
+
+* pathProperty1: ...
+
+### path-id1-activity1
+
+* activityProperty1: ...
+* activityProperty2: ...
+
+### path-id1-activity2
+
+## path-id2
+
+### path-id2-activity1
+
+### path-id2-activity2
+
+### path-id2-activity3
+```
+
+The first-level (`#`) heading is the ID of the skillmap. Under this heading are the map's properties. A skillmap document has one or more _paths_ declared under the second-level (`##`) headings. The path sections contain a set of _activities_ that form the path. The activities for a path are under third-level (`###`) headings.
+
+Each activity section has list of properties. All activities have a `name`, a `description`, and a `type`, along with other properties which are related to the `type` of the section.
+
+```
+### space-cruiser-activity1
+
+* name: Space background
+* description: Create a stellar background image.
+* type: tutorial
+```
+
+### Map properties
+
+The map properties are in the first section of the skillmap document. This starts with a first-level (`#`) heading using the skillmap's name, or its the _map-id_.
+
+* **map-id**: the identification name of the skillmap
+
+```
+# game-maker-guide
+```
+
+Following the heading is a bulleted list of the properties defined for the skillmap.
+
+* **name**: map display name
+* **description**: map description
+* **infoUrl**: ???
+* **backgroundurl**: URL to the background image
+* **bannerurl**: URL to the banner image (displayed in the sidebar when nothing is selected) 
+* **primarycolor**: hex color value for path and locked nodes (slightly translucent)
+* **secondarycolor**: hex color value for unlocked nodes
+* **tertiarycolor**: hex color value for background
+* **highlightcolor**: hex color value for the selected node border and unlocked reward nodes
+* **alternatesources**: ???
```

Comment:
What's this?

Reply:
this is for migrating progress from one skillmap to another. a user's progress (how much they've unlocked, the projects they've created) is tied to the url of the map, so if you change the url you need to add the old one as an alternate source (comma separated, if there are multiple).

we use it here: https://github.com/microsoft/pxt-arcade/blame/master/docs/skillmap/intermediate-skillmap.md#L11

it's formatted like "github:[path to github map]" or "docs:[path to docs map]"